CMakeLists.txt

```diff
@@ -234,6 +235,12 @@ file(GLOB SOURCES
     src/treelearner/*.cpp
 )
 
+if(BUILD_PROFILING_TESTS)
+    # For profiling builds with valgrind/callgrind use -DDEBUG=ON
+    add_executable(lightgbm_profile_single_row_predict profiling/profile_single_row_predict.cpp ${SOURCES} src/c_api.cpp)
+    add_executable(lightgbm_profile_single_row_predict_fast profiling/profile_single_row_predict_fast.cpp ${SOURCES} src/c_api.cpp)
+endif(BUILD_PROFILING_TESTS)
```

Comment:
I'm not sure that we should add profiling part into the repo... @guolinke WDYT?

Reply:
If that is the decision should I also remove the new profiling folder and its contents?

Reply:
> If that is the decision

Let's wait for some other opinions.

> should I also remove the new profiling folder and its contents?

I think yes. Right now `profiling` folder looks some kind of unnatural here. I believe we should transform it into a test during the work on #261 or integrate it into other benchmarks (https://github.com/guolinke/boosting_tree_benchmarks).

Reply:
gently ping @guolinke for your opinion

Reply:
IMO, it is better to remove this in the official repo.

Reply:
But I can keep a branch for the profiling

Reply:
@guolinke I believe that a separate repo (just like for benchmarks) will be more intuitive.

@AlbertoEAF OK, seems that we got a decision. Please exclude profiling from this PR, but also save the corresponding code for some time somewhere in your fork.

Reply:
Thanks for the input :)

I just removed it and kept it in a branch in my fork.

Reply:
Ah, I've just come the thought that you can simply post the content of profiling files in comment for this PR or original issue. 🙂